control_plane/storcon_cli/src/main.rs

```diff
@@ -0,0 +1,643 @@
+                    Method::PUT,
+                    format!("control/v1/node/{node_id}/config"),
+                    Some(req),
+                )
+                .await?;
+        }
+        Command::Tenants {} => {
+            let resp = storcon_client
+                .dispatch::<(), Vec<TenantDescribeResponse>>(
+                    Method::GET,
+                    "control/v1/tenant".to_string(),
+                    None,
+                )
+                .await?;
+            let mut table = comfy_table::Table::new();
+            table.set_header([
+                "TenantId",
+                "ShardCount",
+                "StripeSize",
+                "Placement",
+                "Scheduling",
+            ]);
+            for tenant in resp {
+                let shard_zero = tenant.shards.into_iter().next().unwrap();
+                table.add_row([
+                    format!("{}", tenant.tenant_id),
+                    format!("{}", shard_zero.tenant_shard_id.shard_count.literal()),
+                    format!("{:?}", tenant.stripe_size),
+                    format!("{:?}", tenant.policy),
+                    format!("{:?}", shard_zero.scheduling_policy),
+                ]);
+            }
+
+            println!("{table}");
+        }
+        Command::TenantPolicy {
+            tenant_id,
+            placement,
+            scheduling,
+        } => {
+            let req = TenantPolicyRequest {
+                scheduling,
+                placement,
+            };
+            storcon_client
+                .dispatch::<_, ()>(
+                    Method::PUT,
+                    format!("control/v1/tenant/{tenant_id}/policy"),
+                    Some(req),
+                )
+                .await?;
+        }
+        Command::ServiceRegister {
+            http_host,
+            http_port,
+            region_id,
+            availability_zone_id,
+            version,
+        } => {
+            let instance_id = http_host.clone();
+            let req = ServiceRegisterRequest {
+                instance_id: instance_id.clone(),
+                // We do not expose postgres protocol, but provide a valid-looking host/port for it
+                host: http_host.clone(),
+                port: 6400,
+                http_host: http_host.clone(),
+                http_port,
+                version,
+                region_id,
+                availability_zone_id,
+                disk_size: 0,
+                instance_type: "".to_string(),
+                register_reason: "Storage Controller Virtual Pageserver".to_string(),
+                is_storage_controller: true,
+            };
+
+            let response = storcon_client
+                .dispatch::<ServiceRegisterRequest, ServiceRegisterResponse>(
+                    Method::POST,
+                    "management/api/v2/pageservers".to_string(),
+                    Some(req),
+                )
+                .await?;
```

Comment:
Unlike other subcommands, this one is targeting the control plane, right?

If you want to have these in the same CLI, I think it would be good to have separate sub-commands for interacting with storage controller and control plane API. Otherwise the user has to know from context whether they need to provide the cplane API endpoint and cplane API JWT or the storage controller endpoint with storage controller JWT.

`storcon_cli cplane register-pageserver|tenant-onboard|...`

`storcon_cli storcon tenant-shard-split|...`

Reply:
This command shouldn't really be needed any more (the deployment registers the controller these days), and it was going to be a maintenance pain as these commands are un-testable without a control plane, so I've just removed service-register and tenant-onboard in 8a46cd00eb5dfa6bf1c82e9025c7b3e7d17535f2